docs/design/segments.md

```diff
@@ -82,10 +82,13 @@ For each row in the list of column data, there is only a single bitmap that has
 
 ## Handling null values
 
-By default, Druid string dimension columns use the values `''` and `null` interchangeably. Numeric and metric columns cannot represent `null` but use nulls to mean `0`. However, Druid provides a SQL compatible null handling mode, which you can enable at the system level through `druid.generic.useDefaultValueForNull`. This setting, when set to `false`, allows Druid to create segments _at ingestion time_ in which the following occurs:
+By default, Druid runs in a SQL compatible null handling mode, which allows Druid to create segments _at ingestion time_ in which the following occurs:
```

Comment:
This section could use some rewording to make it sound more like the SQL-compatible mode is the normal one. Various phrasings throughout the section are written as if legacy mode is normal.

For example, the data structures in SQL-compatible mode are described as "additional" over legacy. It'd be better to describe legacy as _missing_ certain structures rather than SQL-compatible as _adding_ them.

This is a nit, and it doesn't need to block this PR, but I think it would be good to do it the rewords in a follow-on PR.